File: backend/src/main/java/com/library/librarymanagementsystem/service/BookService.java
package com.library.librarymanagementsystem.service;

import com.library.librarymanagementsystem.model.Book;
import com.library.librarymanagementsystem.repository.BookRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class BookService {
    
    @Autowired
    private BookRepository bookRepository;
    
    public List<Book> getAllBooks() {
        return bookRepository.findAll();
    }
    
    public Optional<Book> getBookById(Long id) {
        return bookRepository.findById(id);
    }
    
    public Book createBook(Book book) {
        return bookRepository.save(book);
    }
    
    public Book updateBook(Long id, Book bookDetails) throws RuntimeException {
        Book book = bookRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Book not found with id: " + id));
        
        book.setTitle(bookDetails.getTitle());
        book.setAuthor(bookDetails.getAuthor());
        book.setIsbn(bookDetails.getIsbn());
        book.setGenre(bookDetails.getGenre());
        book.setPublicationYear(bookDetails.getPublicationYear());
        book.setPublisher(bookDetails.getPublisher());
        book.setQuantity(bookDetails.getQuantity());
        book.setAvailabilityStatus(bookDetails.getAvailabilityStatus());
        book.setDescription(bookDetails.getDescription());
        
        return bookRepository.save(book);
    }
    
    public void deleteBook(Long id) throws RuntimeException {
        Book book = bookRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Book not found with id: " + id));
        bookRepository.delete(book);
    }
}

File: frontend/src/api/bookService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/books';

const bookService = {
    getAllBooks: () => {
        return axios.get(API_BASE_URL);
    },

    getBookById: (id) => {
        return axios.get(`${API_BASE_URL}/${id}`);
    },

    createBook: (book) => {
        return axios.post(API_BASE_URL, book);
    },

    updateBook: (id, book) => {
        return axios.put(`${API_BASE_URL}/${id}`, book);
    },

    deleteBook: (id) => {
        return axios.delete(`${API_BASE_URL}/${id}`);
    }
};

export default bookService;



